### DISCUSSION TITLE
How to execute stored procedure with cursor.callproc() many times, for list of input parameters?

### DISCUSSION THREAD

**Discussion by urosdigital** [oracle/python-oracledb#78] [Q&A] [answered]
Hi, is it possible to execute stored procedure using cursor.callproc() but using it in a way similar to cursor.executemany()?

i have a stored procedure which accepts two input arguments, im running it like this:

cursor.callproc('my_stored_procedure', ['P00000012345678','T0000087654321'])

In case i have a list of input parameters for which i need to execute stored procedure, is there some other more efficient way than loop?

input_parameters = [['P00000011345678','T0000088654321'],['P00000011145678','T0000088854321'],['P00000011115678','T0000088884321'],['P00000011111678','T0000088888321']]
for parameters in input_parameters:
    cursor.callproc('my_stored_procedure', parameters)

BR,
Uros

**Comment by anthony-tuininga** [comment, marked as answer]
Yes, you can do this:

```python
cursor.executemany("""
    begin
        my_stored_procedure(:1, :2);
    end;""", input_parameters)
```

The method `cursor.callproc()` effectively generates that anonymous PL/SQL block for you automatically -- so you can simply specify it manually for use in `cursor.executemany()`.

**Reply by urosdigital** [reply]
Thanks @anthony-tuininga for this superfast feedback. Problem solved! :)
